src/tests/getAttributes/getAttributes.test.ts

```diff
@@ -0,0 +1,222 @@
+    );
+    assert.equal(attributesResults.edges[0].cursor,
+      modelToCursor({
+        model: {
+          value: 'brown0',
+        },
+        sort: {
+          value: 1,
+        },
+      }),
+    );
+    assert.equal(attributesResults.pageInfo.startCursor, attributesResults.edges[0].cursor);
+    assert.equal(
+      attributesResults.pageInfo.endCursor,
+      attributesResults.edges[attributesResults.edges.length - 1].cursor,
+    );
+    assert.equal(
+      attributesResults.pageInfo.hasPreviousPage,
+      false,
+    );
+    assert.equal(
+      attributesResults.pageInfo.hasNextPage,
+      true,
+    );
+  });
+
+  it('Should return the last 2 attributes', async () => {
+
+    const attributes = await addTestAttributes(); // tslint:disable-line
+
+    const fromCursor = modelToCursor({
+      model: {
+        value: 'brown7',
+      },
+      sort: {
+        value: 1,
+      },
+    });
+
+    const attributesResults = await service.getAttributes(
+      assign({}, getAttributesOptions, {
+        cursor: fromCursor,
+      }),
+    );
+
+    const TWO = 2;
+    assert.equal(attributesResults.edges.length, TWO);
+    assert.equal(attributesResults.pageInfo.hasNextPage, false);
+    assert.equal(attributesResults.pageInfo.hasPreviousPage, true);
+  });
+
+  it('Should throw error when direction is BACKWARDS and cursor is undefined', async () => {
+    // Add 12 Attributes
+    const attributes = await addTestAttributes(); // tslint:disable-line
+    
+    // Get the first 10 attributes
+    const attributesPromise = service.getAttributes(
+      assign({}, getAttributesOptions, {
+        direction: CursorDirection.BACKWARDS,
+      }),
+    );
+
+    return assertError(NoCursorBackwardsDirection, attributesPromise);
+  });
+
+  it('Should return the previous 2 cursors when direction is BACKWARDS', async () => {
+    // Add 12 Attributes
+    const attributes = await addTestAttributes(); // tslint:disable-line
+
+    // Get the first 10 attributes
+    const attributesResult = await service.getAttributes(
+      assign({}, getAttributesOptions, {
+        cursor: fromFirstCursor,
+        direction: CursorDirection.BACKWARDS,
+      }),
+    );
+
+    const THREE = 3;
+    assert.equal(attributesResult.edges.length, THREE);
+    assert.equal(attributesResult.pageInfo.hasNextPage, true);
+    assert.equal(attributesResult.pageInfo.hasPreviousPage, false);
+  });
+
+  it('should return undefiend cursor, if no attributes', async () => {
+    const attributesResult = await service.getAttributes(
+      assign({}, getAttributesOptions, {
+        limit: 1,
+      }),
+    );
+    assert.equal(attributesResult.pageInfo.endCursor, undefined);
+    assert.equal(attributesResult.pageInfo.startCursor, undefined);
+  });
+
+  it('Should return the previous 1 cursors when limit 1', async () => {
+    const attributes = await addTestAttributes(); // tslint:disable-line
```

Comment:
Why is this assigned to a variable? Can't we just await? I think you've done this in a few places in here.